src/System.IO.Compression/src/System/IO/Compression/DeflateZLib/DeflateStream.cs

```diff
@@ -14,12 +14,12 @@ public partial class DeflateStream : Stream
     {
         private const int DefaultBufferSize = 8192;
 
-        private Stream _stream;
+        private Stream _stream = null!; // fields initialized in init methods called from constructor
```

Comment:
It seems like this one should be nullable and just add the corresponding Debug.Assert after calling `EnsureNotDisposed`

Reply:
This will always been checked for null and init-ed within constuctor or through Init method. And EnsureNotDisposed will make sure it will not be null referenced, no need extra Assert

Reply:
I mean, make it nullable -> `Stream? _stream;` instead of `Stream _stream = null!`. Then you will need to add a `Debug.Assert` or `!` in places where the null check is done inside `EnsureNotDisposed` -- since the null check is done in a helper method, in places where you use `_stream` you will need an assert because the compiler can't track that you checked for null already.

Reply:
I understand what you mean, but for me there is no need to make it nullable as it will not be null until the stream is disposed. EnsureNotDisposed() just making sure its not being used after disposal, not for checking if it set null for any other reason. I don't think if anybody touch this file would set it null for any other reason

Reply:
My point was that if for some reason somebody touched this code or added/optimized/changed whatever, that it forces them to call `EnsureNotDisposed`. I'm not a big fan of using `null!` to avoid using `Debug.Asserts` or `!` when it doesn't affect any other API like a return type or such, but I'm fine leaving as is, I just don't want to introduce this pattern much and make it such a common scenario in our code as it makes it confusing.

Reply:
Understand your point, it could prevent from such very edge case with some overhead, I'm ok with making it nullable, just not much convinced its worth 🤔, any opinion @stephentoub ?